silkworm/silkrpc/http/server.hpp

```diff
@@ -72,6 +73,9 @@ class Server {
     //! The acceptor used to listen for incoming TCP connections
     boost::asio::ip::tcp::acceptor acceptor_;
 
+    //! The list of allowed origins for CORS
+    std::vector<std::string>& allowed_origins_;
```

Comment:
I would keep this by value here to avoid the lifecycle dependency with the caller implied by the usage of reference. The same applies to the constructor argument, which in turn can be passed by value and then moved into the member variable.